### DISCUSSION TITLE
Executable Specification

### DISCUSSION THREAD

**Discussion by mayvazyan** [machine/machine.specifications#425] [Q&A] [answered]
Hello,
I've been using Machine Specification quite a lot with "full" .NET framework. There was a separate console runner which allowed to generate a nice HTML document (specification).

Now (in .NET Core & .NET 5) we are using `dotnet test` to run tests. 

Is it still possible to pass some arguments to generate specification in some form?

Also would be great if there are some way to filter specs to be run. 
For example `dotnet test -i When_using_a_car` will run only tests from the `When_using_a_car` class.
While `dotnet test -e When_using_a_car` will run all tests except tests from the  `When_using_a_car` class.
As I remember standalone console runner had such feature as well.

Thanks,
Michael

**Comment by robertcoltheart** [comment, marked as answer]
The reports feature has languished for some time and doesn't really have the features you're looking for, at least not with `dotnet test` anyway. I'll look at reviving this for `v2.0`. What is your use case for outputting a report?

For filtering, pretty sure this was added recently, see here: https://github.com/machine/machine.specifications.runner.visualstudio/pull/95

**Reply by mayvazyan** [reply]
Robert, thank you for the quick response!

I've been thinking of putting it as `executable specification` into our wiki, so anyone on the team (QA, Manager) can read it.

**Reply by robertcoltheart** [reply]
Understood. This would likely take the same route as xUnit, which is to output the test results to an XML. It would then be up to you to apply an `xslt` or transform the XML to HTML using any of the freely available tools out there.

Edit: Just noticed that xUnit _does_ provide a standard xslt file as well. We can look into this.
